airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlQueryUtils.java

```diff
@@ -185,46 +185,46 @@ public static Map<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair,
 
     final Map<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair, CursorBasedStatus> cursorBasedStatusMap = new HashMap<>();
     streams.forEach(stream -> {
-      try {
-        final String name = stream.getStream().getName();
-        final String namespace = stream.getStream().getNamespace();
-        final String fullTableName =
-            getFullyQualifiedTableNameWithQuoting(namespace, name, quoteString);
-
-        final Optional<CursorInfo> cursorInfoOptional =
-            stateManager.getCursorInfo(new io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair(name, namespace));
-        if (cursorInfoOptional.isEmpty()) {
-          throw new RuntimeException(String.format("Stream %s was not provided with an appropriate cursor", stream.getStream().getName()));
-        }
+      final String name = stream.getStream().getName();
+      final String namespace = stream.getStream().getNamespace();
+      final String fullTableName =
+          getFullyQualifiedTableNameWithQuoting(namespace, name, quoteString);
 
-        LOGGER.info("Querying max cursor value for {}.{}", namespace, name);
-        final String cursorField = cursorInfoOptional.get().getCursorField();
+      final Optional<CursorInfo> cursorInfoOptional =
+          stateManager.getCursorInfo(new AirbyteStreamNameNamespacePair(name, namespace));
+      if (cursorInfoOptional.isEmpty()) {
+        throw new RuntimeException(String.format("Stream %s was not provided with an appropriate cursor", stream.getStream().getName()));
+      }
+      final CursorBasedStatus cursorBasedStatus = new CursorBasedStatus();
+      final Optional<String> maybeCursorField = Optional.ofNullable(cursorInfoOptional.get().getCursorField());
```

Comment:
@stephane-airbyte actually this change by @rodireich should have fixed the issue that we were discussing here

https://github.com/airbytehq/airbyte/pull/39143#issuecomment-2181295059

Noticing the use of `Optional` and `ifPresent` function below which were not checked in the previous version. So I will close that issue.

Reply:
yes. This is indeed fixing it. Thank you

Reply:
Thank you for the discussion that led to this investigation!